util/conditions/setter.go

```diff
@@ -42,31 +42,37 @@ func Set(to Setter, condition *clusterv1.Condition) {
 		return
 	}
 
+	// Check if the new conditions already exists, and change it only if there is a status
+	// transition (otherwise we should preserve the current last transition time)-
 	conditions := to.GetConditions()
-	newConditions := make(clusterv1.Conditions, 0, len(conditions))
 	exists := false
-	condition.LastTransitionTime = metav1.NewTime(time.Now().Truncate(time.Second))
 	for i := range conditions {
 		existingCondition := conditions[i]
 		if existingCondition.Type == condition.Type {
 			exists = true
 			if !hasSameState(&existingCondition, condition) {
-				newConditions = append(newConditions, *condition)
-				continue
+				condition.LastTransitionTime = metav1.NewTime(time.Now().UTC().Truncate(time.Second))
+				conditions[i] = *condition
 			}
+			condition.LastTransitionTime = existingCondition.LastTransitionTime
```

Comment:
This looks a bit odd to me, as we are setting the time in the if statement above, then overwriting it, which doesn't effect the end result but is intended for when the if block isn't entered if I've understood correctly

Perhaps putting it in an else or putting a break at the end of the if statement might make the logic more obvious?

Reply:
good catch it is missing a break in the if and for sure a unit test. I'm going to fix this asap

Reply:
fixed.
there is already a unit test, but as you correctly pointed out, the lack of break wasn't affecting the end result.
However now the code is cleaner, and the statement after the if echoes the LastTransitionTime on the input variable